Formula/k3d.rb

```diff
@@ -19,14 +20,17 @@ class K3d < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "e117dd2b6ebf56a0def38e790920f5c9ac9bbdbfe1abb617c933f46db8770f40"
   end
 
+  depends_on "gnu-sed" => :build
   depends_on "go" => :build
 
+  # curl and sed used to fetch/parse latest stable K3s version
+  uses_from_macos "curl" => :build
+
   def install
-    system "go", "build",
-           "-mod", "vendor",
-           "-ldflags", "-s -w -X github.com/rancher/k3d/v#{version.major}/version.Version=v#{version}"\
-                       " -X github.com/rancher/k3d/v#{version.major}/version.K3sVersion=latest",
-           "-trimpath", "-o", bin/"k3d"
+    # Build k3d
+    system  "make", "build", # Makefile includes fetching the latest stable K3s version
```

Comment:
What's wrong with the current way of building this?

Reply:
Two things:
1. It's effectively just the exact same commands that are already included in the Makefile (why rewrite it here and having to change it in two places in case anything changes? :man_shrugging: )
2. The Makefile fills the K3s version variable by doing some curl + sed magic (would require even more lines in the formula otherwise)

Reply:
Here are a few reasons why:
1. Git clones are slow and expensive, and it really sucks to have to use them just to fetch version information.
2. The build is more transparent when it's not hidden behind a `Makefile`. That's useful for users reading the formula who just want to understand what goes on when something gets built. (It's not always possible, since some `Makefiles` are too complicated, but Go builds are usually simple enough it's not too bad to do it all in the formula.)
3. It's probably pretty useful to have the version information generated from a Homebrew installation be different from other installation methods. That is, unless our builds are byte-for-byte identical to other installations, but I doubt they are (or that they'll continue to be if they are at the moment). This simplifies tracking down bugs, for example.

Reply:
> It's effectively just the exact same commands that are already included in the Makefile (why rewrite it here and having to change it in two places in case anything changes? 🤷‍♂️ )

Because we can optimize this more than we can a Makefile.

> The Makefile fills the K3s version variable by doing some curl + sed magic (would require even more lines in the formula otherwise)

Curl + sed magic does not sound very stable or reliable, while our  variables are.

Reply:
Okay, so no problem for me to keep the go build logic in here if you think it's the better way to go and I don't think we'll have many more variable injections in the future.

With regards to the curl + sed magic: were just fetching a file from the K3s channel server and parse it to get the latest tag of the stable channel so we can inject the K3s version variable (latest is not good here).
So it's not about the k3d version (which we use from homebrew), but about K3s which is used by k3d, so we have to use curl+sed (or a more "stable" way with jq, but that's not installed by default on many systems (or some ruby if that's possible)).

Reply:
Ruby can parse JSON natively, so it should be a lot easier than with `sed`.

Reply:
We're slowly requiring `std_go_args` to be used for all Go dependents. It should have been used here before.

Makefiles are fine in general, but they typically make it impossible to use our `std_go_args` or make it complicated enough that it's not really any simpler using the Makefile.

Looking at the Makefile here for example, it looks like it is possible to override but not append to LDFLAGS, so we would have to handle the versioning logic ourselves anyway.

Reply:
Alright, then I'll go with the new standards here :+1: